core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/PrefixDeclProcessor.java

```diff
@@ -71,8 +86,18 @@ public static Map<String, String> process(ASTOperationContainer qc) throws Malfo
 			prefixMap.put(prefix, iri);
 		}
 
+		int preDefaultPrefixes = 0;
+
+		// insert the default prefixes if presents
+		if (customDefaultPrefixes != null && !customDefaultPrefixes.isEmpty()) {
```

Comment:
This `if` statement could be removed if the argument could be made non-nullable in favor of an empty map, then the following `for` loop would simple do nothing if the map were empty.